### PR TITLE
Feature: Share safe app

### PR DESCRIPTION
**Pull request by mikhailxyz**
## What it solves

Resolves #

### CODE REVIEW

File: src/components/common/ChainIndicator/index.tsx [hunk 1]
```diff
@@ -26,6 +26,8 @@ const ChainIndicator = ({ chainId, className, inline = false }: ChainIndicatorPr
     }
   }, [chainConfig])
 
+  if (!chainConfig?.chainName) return null
```
Comment (mikhailxyz) [inline src/components/common/ChainIndicator/index.tsx:29]: The change here is needed for the situation where an app is supported on a chain we don't support to prevent spending computational resources on filtering. Currently, if you pass an unsupported chain id to this component, it renders a space

File: src/components/create-safe/status/hooks/useWatchSafeCreation.ts [hunk 2]
```diff
@@ -35,8 +38,21 @@ const useWatchSafeCreation = ({
 
     if (status === SafeCreationStatus.INDEXED) {
       trackEvent(CREATE_SAFE_EVENTS.GET_STARTED)
+      const chainPrefix = chain?.shortName
```
Comment (mikhailxyz) [inline src/components/create-safe/status/hooks/useWatchSafeCreation.ts:41]: I added the chain prefix to the redirect to fix a bug where if you refresh the app using an URL with a non-EIP3770 address the app shows a 404 error

File: src/components/common/ChainIndicator/index.tsx [hunk 3]
```diff
@@ -26,6 +26,8 @@ const ChainIndicator = ({ chainId, className, inline = false }: ChainIndicatorPr
...
+
   return (
     <span style={style} className={classnames(inline ? css.inlineIndicator : css.indicator, className)}>
       {chainConfig?.chainName || ' '}
```
Comment (iamacook) [inline src/components/common/ChainIndicator/index.tsx:33]: We can remove the fallback here now that it returns `null`.

File: src/components/create-safe/status/hooks/useWatchSafeCreation.ts [hunk 4]
```diff
@@ -4,24 +4,27 @@ import { useRouter } from 'next/router'
...
   setPendingSafe,
   setStatus,
+  chainId,
 }: {
   status: SafeCreationStatus
   safeAddress: string | undefined
   pendingSafe: PendingSafeData | undefined
   setPendingSafe: Dispatch<SetStateAction<PendingSafeData | undefined>>
   setStatus: Dispatch<SetStateAction<SafeCreationStatus>>
+  chainId: string
```
Comment (iamacook) [inline src/components/create-safe/status/hooks/useWatchSafeCreation.ts:24]: It's possible to use `useChainId` instead of passing it.

File: src/components/safe-apps/SafeAppLandingPage/UseApp.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+type Props = {
+  appUrl: string
+  wallet: ConnectedWallet | null
+  onConnectWallet: () => Promise<void>
+  safes: string[]
+  chainId: string
+  chainPrefix?: string
+}
+
+const UseApp = ({ wallet, onConnectWallet, safes, chainId, chainPrefix, appUrl }: Props): React.ReactElement => {
```
Comment (iamacook) [inline src/components/safe-apps/SafeAppLandingPage/UseApp.tsx:22]: I feel like this could be easily confused with a hook. I'd suggest `TryApp` or something similar instead.

File: src/components/safe-apps/SafeAppLandingPage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+const SafeAppLanding = ({ appUrl, chainId }: Props) => {
+  const { safeApp, isLoading } = useSafeAppFromManifest(appUrl, chainId)
+  const [backendApp, , backendAppLoading] = useSafeAppFromBackend(appUrl)
+  const wallet = useWallet()
+  const onboard = useOnboard()
+  const ownedSafes = useOwnedSafes()[chainId] ?? []
+  const chain = useAppSelector((state) => selectChainById(state, chainId))
+  // show demo if the app was shared for mainnet or we can find the mainnet chain id on the backend
+  const showDemo = chainId === '1' || !!backendApp?.chainIds.includes('1')
```
Comment (iamacook) [inline src/components/safe-apps/SafeAppLandingPage/index.tsx:32]: I think we should move the `1` out into a constant as this is easily overlooked.

File: src/components/create-safe/status/hooks/useWatchSafeCreation.ts (hunk 4 above)
Comment (mikhailxyz) [reply src/components/create-safe/status/hooks/useWatchSafeCreation.ts:24]: > It's possible to use `useChainId` instead of passing it.

Why? The place where the hook is used already has chainId in scope. Plus, with an explicit parameter, it's easier to test than having to mock internals

Comment (iamacook) [reply src/components/create-safe/status/hooks/useWatchSafeCreation.ts:24]: I didn't realise it's just used the once.

File: src/hooks/safe-apps/useChainIdQueryParam.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import { useMemo } from 'react'
+
+type ReturnType = {
+  chainId?: string
+  validChainId: boolean
+  loading: boolean
+  error: string | undefined
+}
+
+const useChainIdQueryParam = (): ReturnType => {
```
Comment (usame-algan) [inline src/hooks/safe-apps/useChainIdQueryParam.ts:12]: Is there a reason we are not reusing the `chain` query param e.g. `?chain=eth`?

When I try to access a share url like `share/safe-app?appUrl=https://apps.gnosis-safe.io/drain-safe&chainId=1` it still display Rinkeby as the selected network.

Comment (mikhailxyz) [reply src/hooks/safe-apps/useChainIdQueryParam.ts:12]: I wondered if it's possible to switch the network and somehow didn't find this param. Will adjust the share link now

File: src/hooks/safe-apps/useSafeAppFromBackend.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import useAsync, { AsyncResult } from '../useAsync'
+import { trimTrailingSlash } from '@/utils/url'
+
+const useSafeAppFromBackend = (url: string, chainId: string): AsyncResult<SafeAppData> => {
+  const [backendApp, error, loading] = useAsync(async () => {
+    // We do not have a single standard for storing URLs, it may be stored with or without a trailing slash.
+    // But for the request it has to be an exact match.
+    const retryUrl = url.endsWith('/') ? trimTrailingSlash(url) : `${url}/`
+    let response = await getSafeApps(chainId, { url })
+    if (!response?.[0]) {
```
Comment (usame-algan) [inline src/hooks/safe-apps/useSafeAppFromBackend.ts:13]: The return type for `response` suggests that it can't be undefined so we should be able to do `response[0]`.

Also for the future, I think we should consider handling this logic in the gateway so that we only have to do one request. Or alternatively, standardize all safe apps urls.

File: src/components/safe-apps/AppCard/index.tsx [hunk 9]
```diff
@@ -92,10 +92,19 @@ const AppCardContainer = ({ url, children, variant }: AppCardContainerProps): Re
...
       <img src={safeApp.iconUrl} style={{ width: 52, height: 52 }} alt={`${safeApp.name} logo`} />
+      <IconButton
+        aria-label={`Share ${safeApp.name}`}
+        size="small"
+        onClick={onShareClick}
+        title={`Copy share URL for ${safeApp.name}`}
+        sx={{ width: '32px', position: 'absolute', top: 2, left: 2 }}
+      >
+        <ShareIcon width={16} alt="Share icon" />
+      </IconButton>
```
Comment (usame-algan) [inline src/components/safe-apps/AppCard/index.tsx:107]: We are missing a UI feedback when the user presses the button. In safe-react we had a notification pop up but I could also imagine handling it the same way we do when copying the safe address from the sidebar by showing a tooltip. Wdyt?

File: src/components/safe-apps/SafeAppLandingPage/index.tsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+const SafeAppLanding = ({ appUrl, chainId }: Props) => {
+  const { safeApp, isLoading } = useSafeAppFromManifest(appUrl, chainId)
+  const [backendApp, , backendAppLoading] = useSafeAppFromBackend(appUrl, chainId)
+  const wallet = useWallet()
+  const onboard = useOnboard()
+  const ownedSafes = useOwnedSafes()[chainId] ?? []
+  const chain = useAppSelector((state) => selectChainById(state, chainId))
+  // show demo if the app was shared for mainnet or we can find the mainnet chain id on the backend
+  const showDemo = chainId === CHAIN_ID_WITH_A_DEMO || !!backendApp?.chainIds.includes(CHAIN_ID_WITH_A_DEMO)
+  console.log({ safeApp, backendApp, isLoading })
```
Comment (usame-algan) [inline src/components/safe-apps/SafeAppLandingPage/index.tsx:35]:
```suggestion

```

File: src/hooks/safe-apps/useSafeAppFromBackend.ts (hunk 8 above)
Comment (mikhailxyz) [reply src/hooks/safe-apps/useSafeAppFromBackend.ts:13]:
```
export type AsyncResult<T> = [result: T | undefined, error: Error | undefined, loading: boolean]
```

The undefined comes from the hook

Comment (usame-algan) [reply src/hooks/safe-apps/useSafeAppFromBackend.ts:13]: > The undefined comes from the hook

I meant that its possible to do `response[0]` instead of `response?.[0]` since `response` itself can't be undefined.

File: src/components/safe-apps/AppCard/index.tsx (hunk 9 above)
Comment (usame-algan) [reply src/components/safe-apps/AppCard/index.tsx]: Nice, the tooltip works great! I think in the process the share button got oval shaped though:

Screenshot 2022-09-09 at 12 02 49

File: src/hooks/safe-apps/useSafeAppFromBackend.ts (hunk 8 above)
Comment (mikhailxyz) [reply src/hooks/safe-apps/useSafeAppFromBackend.ts]: Ah, you're right. Fixed it now

File: src/components/safe-apps/SafeAppLandingPage/index.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    trackEvent({ ...SAFE_APPS_EVENTS.SHARED_APP_OPEN_DEMO, label: appUrl })
+  }
+
+  if (isLoading || backendAppLoading) {
+    return <div>Loading...</div>
+  }
+
+  if (!safeApp) {
+    return <div>No safe app found</div>
+  }
```
Comment (usame-algan) [inline src/components/safe-apps/SafeAppLandingPage/index.tsx:67]: For some apps we display this message even though the app exists e.g. `share/safe-app?appUrl=https://apy.plasma.finance&chainId=1&chain=eth`

Should we extend the condition to `!safeApp && !backendApp`?

File: src/components/safe-apps/AppCard/index.tsx [hunk 12]
```diff
@@ -117,19 +127,13 @@ const CompactAppCard = ({ url, safeApp, onPin, pinned }: CompactSafeAppCardProps
 
 const AppCard = ({ safeApp, pinned, onPin, onDelete, variant = 'default' }: AppCardProps): ReactElement => {
   const router = useRouter()
-  const chainId = useChainId()
+  const currentChain = useCurrentChain()
 
-  const shareUrl = `${SAFE_REACT_URL}/share/safe-app?appUrl=${safeApp.url}&chainId=${chainId}`
+  const shareUrl = `${window.location.origin}${AppRoutes.share.safeApp}?appUrl=${safeApp.url}&chainId=${currentChain?.chainId}&chain=${currentChain?.shortName}`
```
Comment (usame-algan) [inline src/components/safe-apps/AppCard/index.tsx:132]: Do we need both `chainId` and `chain` as query params or is `chain` sufficient?

File: src/components/safe-apps/SafeAppLandingPage/index.tsx (hunk 11 above)
Comment (mikhailxyz) [reply src/components/safe-apps/SafeAppLandingPage/index.tsx:67]: The manifest.json is a hard requirement, so it's the source of truth for the existence of the app

https://docs.gnosis-safe.io/build/sdks/safe-apps/releasing-your-safe-app

**Comment by yagopv** [comment]
Hey @mikhailxyz any reason to keep the sidebar? I like more the wider design in the current web without the sidebar

image

**Comment by mikhailxyz** [comment]
@yagopv No, didn't pay attention to it. Will try to hide t

**Comment by yagopv** [comment]
Notice some differences from the old one:

1) Guess you added the App URL on purpose, not sure, but I like it there 🤗
2) What do you think about having the warning going up to fill the empty space as now we are not showing the demo safe if not available in the network? ... Or the available networks going to the right or something that clears the empty space

image

**Comment by mikhailxyz** [comment]
> Guess you added the App URL on purpose, not sure, but I like it there 🤗

It was in the design

> What do you think about having the warning going up to fill the empty space as now we are not showing the demo safe if not available in the network? ... Or the available networks going to the right or something that clears the empty space

I don't have a preference, but I'd check this with the design team first

File: src/components/safe-apps/AppCard/index.tsx (hunk 12 above)
Comment (mikhailxyz) [reply src/components/safe-apps/AppCard/index.tsx]: I replaced it in favor of `chain`

File: src/components/safe-apps/AppCard/index.tsx (hunk 9 above)
Comment (mikhailxyz) [reply src/components/safe-apps/AppCard/index.tsx]: Fixed now

**Comment by mikhailxyz** [comment]
Something bad happened to tests when I switch from `chainId` to `chain`, they run fine locally but here they time out

File: src/components/safe-apps/SafeAppLandingPage/index.tsx (hunk 11 above)
Comment (mikhailxyz) [reply src/components/safe-apps/SafeAppLandingPage/index.tsx:65]: We need to check if the app is in the default list, and this is only possible by fetching the app from the backend. Also, the information about supported chains is only available on the backend

File: src/components/common/ChainIndicator/index.tsx [hunk 13]
```diff
@@ -26,9 +26,11 @@ const ChainIndicator = ({ chainId, className, inline = false }: ChainIndicatorPr
     }
   }, [chainConfig])
 
+  if (!chainConfig?.chainName) return null
```
Comment (katspaugh) [inline src/components/common/ChainIndicator/index.tsx:29]: The idea was to render the component even if the chains are still loading.
I would suggest to restore it how it was. If you want to improve it, maybe replace the `|| ' '` with a Skeleton?

File: src/components/safe-apps/AppCard/index.tsx [hunk 14]
```diff
@@ -92,10 +94,18 @@ const AppCardContainer = ({ url, children, variant }: AppCardContainerProps): Re
...
 }
 
-const CompactAppCard = ({ url, safeApp, onPin, pinned }: CompactSafeAppCardProps): ReactElement => (
+const CompactAppCard = ({ url, safeApp, onPin, pinned, shareUrl }: CompactSafeAppCardProps): ReactElement => (
   <AppCardContainer url={url} variant="compact">
     <div className={styles.compactCardContainer}>
       <img src={safeApp.iconUrl} style={{ width: 52, height: 52 }} alt={`${safeApp.name} logo`} />
+      <CopyButton
+        text={shareUrl}
+        ariaLabel={`Share ${safeApp.name}`}
```
Comment (katspaugh) [inline src/components/safe-apps/AppCard/index.tsx:103]: Does the `ariaLabel` have to be different than the text? I would remove this prop and just use `initialToolTipText` for both.

File: src/components/common/PageLayout/index.tsx [hunk 15]
```diff
@@ -29,14 +31,14 @@ const PageLayout = ({ children }: { children: ReactElement }): ReactElement => {
...
-      <aside className={css.sidebar}>{sidebar}</aside>
+      {showSidebar && <aside className={css.sidebar}>{sidebar}</aside>}
 
       {/* Mobile sidebar */}
       <Drawer variant="temporary" anchor="left" open={isMobileDrawerOpen} onClose={onMenuToggle}>
         {sidebar}
       </Drawer>
 
-      <div className={css.main}>
+      <div className={`${css.main} ${!showSidebar ? css.mainNoSidebar : ''}`}>
```
Comment (katspaugh) [inline src/components/common/PageLayout/index.tsx:41]: Please use `classnames`.
E.g. `className={classnames(css.main, {[css.mainNoSidebar]: !showSidebar })}`

File: src/components/common/PageLayout/index.tsx [hunk 16]
```diff
@@ -7,10 +7,12 @@ import Header from '@/components/common//Header'
 import css from './styles.module.css'
 import SafeLoadingError from '../SafeLoadingError'
 import Footer from '../Footer'
+import { AppRoutes } from '@/config/routes'
 
 const PageLayout = ({ children }: { children: ReactElement }): ReactElement => {
   const router = useRouter()
   const [isMobileDrawerOpen, setIsMobileDrawerOpen] = useState<boolean>(false)
+  const showSidebar = !router.pathname.includes(AppRoutes.share.safeApp)
```
Comment (katspaugh) [inline src/components/common/PageLayout/index.tsx:15]: You could probably do a strict comparison here?
```
router.pathname === AppRoutes.share.safeApp
```

File: src/components/safe-apps/SafeAppLandingPage/index.tsx (hunk 11 above)
Comment (usame-algan) [reply src/components/safe-apps/SafeAppLandingPage/index.tsx:65]: Thanks for explaining, I think I understand it now!

File: src/components/create-safe/status/hooks/useWatchSafeCreation.ts [hunk 17]
```diff
@@ -35,8 +38,26 @@ const useWatchSafeCreation = ({
...
+        const address = `${chainPrefix}:${safeAddress}`
+        const redirectUrl = router.query?.safeViewRedirectURL
+        if (typeof redirectUrl === 'string') {
+          // We're prepending the safe address directly here because the `router.push` doesn't parse
+          // The URL for already existing query params
+          const hasQueryParams = redirectUrl.includes('?')
+          const appendChar = hasQueryParams ? '&' : '?'
+
+          if (redirectUrl.includes('apps')) {
+            trackEvent({ ...SAFE_APPS_EVENTS.SHARED_APP_OPEN_AFTER_SAFE_CREATION })
```
Comment (katspaugh) [inline src/components/create-safe/status/hooks/useWatchSafeCreation.ts:53]: Can we track this on the share page instead?

File: src/hooks/safe-apps/useChainFromQueryParams.ts [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+import useChains from '@/hooks/useChains'
+
+type ReturnType = {
+  chain?: ChainInfo
+  validChain: boolean
+  loading: boolean
+  error: string | undefined
+}
+
+const useChainFromQueryParams = (): ReturnType => {
```
Comment (katspaugh) [inline src/hooks/safe-apps/useChainFromQueryParams.ts:13]: How is this different from the existing `useChainId`?

File: src/hooks/useLastSafe.ts [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+import chains from '@/config/chains'
+
+const useLastSafe = (): string | undefined => {
+  const chainId = useChainId()
+  const lastSafeAddress = useAppSelector((state) => selectLastSafeAddress(state, chainId))
+  const prefix = Object.keys(chains).find((prefix) => chains[prefix] === chainId)
+  return prefix && lastSafeAddress ? `${prefix}:${lastSafeAddress}` : undefined
+}
+
+export { useLastSafe }
```
Comment (katspaugh) [inline src/hooks/useLastSafe.ts:13]: Why not default?

File: src/pages/share/safe-app.tsx [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+import { useChainFromQueryParams } from '@/hooks/safe-apps/useChainFromQueryParams'
+import { SafeAppLanding } from '@/components/safe-apps/SafeAppLandingPage'
+
+const ShareSafeApp = () => {
+  const router = useRouter()
+  const [appUrl, routerReady] = useSafeAppUrl()
+  const { chain, validChain, loading: chainLoading, error: chainError } = useChainFromQueryParams()
+
+  if (!routerReady || chainLoading) {
+    return null
```
Comment (katspaugh) [inline src/pages/share/safe-app.tsx:12]: Can't we show at least something?

File: src/pages/share/safe-app.tsx [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  if (!appUrl || !validChain || !chain) {
+    router.push('/')
```
Comment (katspaugh) [inline src/pages/share/safe-app.tsx:16]: `AppRoutes.index`

File: src/tests/test-utils.tsx [hunk 22]
```diff
@@ -30,30 +31,42 @@ const mockRouter = (props: Partial<NextRouter> = {}): NextRouter => ({
...
+const getProviders: (options: {
+  routerProps?: Partial<NextRouter>
+  initialReduxState?: Partial<RootState>
+}) => React.FC<{ children: React.ReactElement }> = ({ routerProps, initialReduxState }) =>
   function ProviderComponent({ children }) {
     const { StoreHydrator } = require('@/store') // require dynamically to reset the store
 
     return (
-      <StoreHydrator>
+      <StoreHydrator initialState={initialReduxState}>
```
Comment (katspaugh) [inline src/tests/test-utils.tsx:42]: I would mock localStorage instead. That way we would also test that hydration from LS works.

**Review by usame-algan — APPROVED** [review]
👍

File: src/components/create-safe/status/hooks/useWatchSafeCreation.ts (hunk 17 above)
Comment (mikhailxyz) [reply src/components/create-safe/status/hooks/useWatchSafeCreation.ts:53]: How? We don't go back to the share page after the safe creation

File: src/tests/test-utils.tsx (hunk 22 above)
Comment (mikhailxyz) [reply src/tests/test-utils.tsx:42]: It still seems to work because a couple of tests depend on local storage persistence. I'm not sure if hydration is relevant here because, AFAIK, we plan to export the app to a static HTML

File: src/hooks/useLastSafe.ts (hunk 19 above)
Comment (mikhailxyz) [reply src/hooks/useLastSafe.ts:13]: I'm not a big fan of default exports for reasons listed [here](https://github.com/basarat/typescript-book/blob/master/docs/tips/defaultIsBad.md)

File: src/hooks/safe-apps/useChainFromQueryParams.ts (hunk 18 above)
Comment (mikhailxyz) [reply src/hooks/safe-apps/useChainFromQueryParams.ts:13]: `useChainId` doesn't guarantee that it will return the chain id from the query param but the user flow is dependent on `chainId` from the query params

File: src/components/common/ChainIndicator/index.tsx (hunk 13 above)
Comment (mikhailxyz) [reply src/components/common/ChainIndicator/index.tsx:29]: >  If you want to improve it, maybe replace the || ' ' with a Skeleton?

That wouldn't help in our scenario. Our problem with the component was that if you pass a chainId that we don't support, it will render an empty but occupying space component. Out of curiosity, why does the whitespace need to be rendered if the chains are loading?

I think if it's a hard requirement on your side, we might be better with a separate component then

File: src/tests/test-utils.tsx (hunk 22 above)
Comment (iamacook) [reply src/tests/test-utils.tsx:42]: "hydration" related to store population here. It's not related to NextJS hydration.

File: src/hooks/useLastSafe.ts (hunk 19 above)
Comment (katspaugh) [reply src/hooks/useLastSafe.ts:13]: Most of our hooks and other modules use a default export, so I'd prefer if you follow that pattern.

File: src/hooks/safe-apps/useChainFromQueryParams.ts (hunk 18 above)
Comment (katspaugh) [reply src/hooks/safe-apps/useChainFromQueryParams.ts:13]: What does `useCurrentChain` lack from your requirements? It defaults to the last chainId if no query present. Isn't that good for the share page as well?

File: src/components/common/ChainIndicator/index.tsx (hunk 13 above)
Comment (katspaugh) [reply src/components/common/ChainIndicator/index.tsx:29]: If you pass an invalid chainId, you've got bigger problems in the parent component. That's why I suggest to use `useCurrentChain` instead of your new `useChainFromQuery`. It falls back to the last chain when no valid chainId is passed.

File: src/hooks/safe-apps/useChainFromQueryParams.ts (hunk 18 above)
Comment (mikhailxyz) [reply src/hooks/safe-apps/useChainFromQueryParams.ts:13]: That's open for discussion in the future. With is the current implemented in `safe-react` the app is shared for a specific chain

Comment (mikhailxyz) [reply src/hooks/safe-apps/useChainFromQueryParams.ts:13]: If the app doesn't exist on last chainId, it's not really good

File: src/components/common/ChainIndicator/index.tsx (hunk 13 above)
Comment (mikhailxyz) [reply src/components/common/ChainIndicator/index.tsx:29]: > If you pass an invalid chainId, you've got bigger problems in the parent component

What do you mean? On the config service, we store all chains the app supports, not only the ones we support. Of course we can filter them out but to me it seems like a waste of resources

Comment (mikhailxyz) [reply src/components/common/ChainIndicator/index.tsx:29]: I explained why `useCurrentChain` doesn't suit us in the other thread. 

tl;dr the app is shared per specific chain and if no query is present, the share link is invalid

File: src/components/create-safe/status/hooks/useWatchSafeCreation.ts (hunk 17 above)
Comment (mikhailxyz) [reply src/components/create-safe/status/hooks/useWatchSafeCreation.ts:53]: To the safe + app page directly

File: src/components/common/ChainIndicator/index.tsx (hunk 13 above)
Comment (katspaugh) [reply src/components/common/ChainIndicator/index.tsx:29]: I see now, I thought you meant invalid chain as in the Safe doesn't support it.

File: src/components/common/PageLayout/index.tsx (hunk 15 above)
Comment (mikhailxyz) [reply src/components/common/PageLayout/index.tsx:41]: :+1: I didn't see it was there, btw there's a classnames alternative with a smaller bundle footprint, we might want to switch to that: https://www.npmjs.com/package/clsx

File: src/components/common/ChainIndicator/index.tsx (hunk 13 above)
Comment (mikhailxyz) [reply src/components/common/ChainIndicator/index.tsx]: I restored the previous state but added a flag not to render the whitespace if the chain isn't found
